Library/Formula/erlang.rb

```diff
@@ -39,11 +39,13 @@ class Erlang < Formula
   depends_on "autoconf" => :build
   depends_on "automake" => :build
   depends_on "libtool" => :build
-  depends_on "openssl"
   depends_on "unixodbc" if MacOS.version >= :mavericks
   depends_on "fop" => :optional # enables building PDF docs
   depends_on "wxmac" => :recommended # for GUI apps like observer
 
+  depends_on "libressl" => :optional
+  depends_on "openssl" unless build.with? "libressl"
```

Comment:
> But it won't use OpenSSL during the compile of the formula itself. 

Of course it won't, you can't use two libraries with the same API without breaking the build.

> We know the :recommended and :optional dance on LibreSSL/OpenSSL is imperfect, but LibreSSL isn't yet popular enough to merit rewriting a chunk of Homebrew's core code to handle the use case.

I've provided you with a snippet that solves the problem.
Apparently you don't like conditions in recipes.

I am discouraged to see that you prefer code "beauty" over correct behaviour.

What you are proposing is semantically incorrect. It is a cludge. Homebrew shouldn't try to get OpenSSL if it is not needed. Moreover, this way you can break your build by specifying:

```
brew install erlang --without-openssl
```

I don't want to spend any more of your time, so if you don't agree with me, just close [my pull request](https://github.com/Homebrew/homebrew/pull/37878). I had some manually edited recipes and now I see I can easily write a script to automate this. So if you don't like this contribution, I don't really care.